tests/blazesym.rs

```diff
@@ -164,45 +164,47 @@ fn lookup_dwarf_no_debug_info() {
-        .join("data")
-        .join("libtest-so.so");
-
-    let so_cstr = CString::new(test_so.clone().into_os_string().into_vec()).unwrap();
-    let handle = unsafe { libc::dlopen(so_cstr.as_ptr(), libc::RTLD_NOW) };
-    assert!(!handle.is_null());
-
-    let the_answer_addr = unsafe { libc::dlsym(handle, "the_answer\0".as_ptr().cast()) };
-    assert!(!the_answer_addr.is_null());
-
-    let normalizer = Normalizer::new();
-    let norm_addrs = normalizer
-        .normalize_user_addrs_sorted([the_answer_addr as Addr].as_slice(), Pid::Slf)
-        .unwrap();
-    assert_eq!(norm_addrs.addrs.len(), 1);
-    assert_eq!(norm_addrs.meta.len(), 1);
-
-    let rc = unsafe { libc::dlclose(handle) };
-    assert_eq!(rc, 0, "{}", Error::last_os_error());
-
-    let norm_addr = norm_addrs.addrs[0];
-    let meta = &norm_addrs.meta[norm_addr.1];
-    assert_eq!(meta.binary().unwrap().path, test_so);
-
-    let cfg = SymbolSrcCfg::Elf(cfg::Elf {
-        file_name: test_so,
-        // TODO: Fix our symbolizer. Base address should be 0.
-        base_address: 0x1000,
-    });
-    let symbolizer = BlazeSymbolizer::new().unwrap();
-    let results = symbolizer
-        .symbolize(&cfg, &[norm_addr.0])
-        .unwrap()
-        .into_iter()
-        .flatten()
-        .collect::<Vec<_>>();
-    assert_eq!(results.len(), 1);
+    fn test(so: &str) {
+        let test_so = Path::new(&env!("CARGO_MANIFEST_DIR")).join("data").join(so);
+        let so_cstr = CString::new(test_so.clone().into_os_string().into_vec()).unwrap();
+        let handle = unsafe { libc::dlopen(so_cstr.as_ptr(), libc::RTLD_NOW) };
+        assert!(!handle.is_null());
+
+        let the_answer_addr = unsafe { libc::dlsym(handle, "the_answer\0".as_ptr().cast()) };
+        assert!(!the_answer_addr.is_null());
+
+        let normalizer = Normalizer::new();
+        let norm_addrs = normalizer
+            .normalize_user_addrs_sorted([the_answer_addr as Addr].as_slice(), Pid::Slf)
+            .unwrap();
+        assert_eq!(norm_addrs.addrs.len(), 1);
+        assert_eq!(norm_addrs.meta.len(), 1);
+
+        let rc = unsafe { libc::dlclose(handle) };
+        assert_eq!(rc, 0, "{}", Error::last_os_error());
+
+        let norm_addr = norm_addrs.addrs[0];
+        let meta = &norm_addrs.meta[norm_addr.1];
+        assert_eq!(meta.binary().unwrap().path, test_so);
+
+        let cfg = SymbolSrcCfg::Elf(cfg::Elf {
+            file_name: test_so,
+            // TODO: Fix our symbolizer. Base address should be 0.
+            base_address: 0x1000,
+        });
+        let symbolizer = BlazeSymbolizer::new().unwrap();
+        let results = symbolizer
+            .symbolize(&cfg, &[norm_addr.0])
+            .unwrap()
+            .into_iter()
+            .flatten()
+            .collect::<Vec<_>>();
+        assert_eq!(results.len(), 1);
+
+        let result = results.first().unwrap();
+        assert_eq!(result.symbol, "the_answer");
+    }
```

Comment:
after dlclose we can do all the same normalization steps for the same address, but ensure that we get Unknown meta? This will test that we don't find some random stuff and we don't crash. WDYT?

Reply:
We already have a test for that: https://github.com/libbpf/blazesym/blob/f651864a8a4faa2c7b24199c8b42eb4e531a6137/src/normalize/normalize.rs#L402-L415

I want to avoid overloading this one with more logic, to be honest; it's already pretty long.